### PR TITLE
[Messaging Extensions] Add Client Options Sample

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to add samples for the Event Hubs and Service Bus Functions extensions packages demonstrating how to configure the client options
not offered in the `host.json` settings.

### CODE REVIEW

File: eng/Packages.Data.props [hunk 1]
```diff
@@ -170,7 +170,7 @@
   <ItemGroup Condition="('$(IsTestProject)' == 'true') OR ('$(IsTestSupportProject)' == 'true') OR ('$(IsPerfProject)' == 'true') OR ('$(IsStressProject)' == 'true') OR ('$(IsSamplesProject)' == 'true')">
     <PackageReference Update="ApprovalTests" Version="3.0.22" />
     <PackageReference Update="ApprovalUtilities" Version="3.0.22" />
-    <PackageReference Update="Azure.Messaging.EventHubs" Version="5.2.0" />
+    <PackageReference Update="Azure.Messaging.EventHubs" Version="5.6.2" />
```
Comment (jsquire) [inline eng/Packages.Data.props:173]: Orthogonal, but we should not be basing tests on a year-old version.   Bumping to the most recent GA.

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/README.md [hunk 2]
```diff
@@ -217,6 +217,39 @@ public static async Task Run(
 }
 ```
 
+### Configuring the ServiceBusClient
+
+The `ServiceBusClient` can be configured using the  [settings](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-service-bus#additional-settings-for-version-5x) available in `host.json` for the majority of scenarios.  Some less common scenarios, however, require configuring options not available in the settings.  For example, when hosting your Function locally, it is sometimes necessary to specify the web sockets transport to request that the client use port 443 rather than the default 5671 and 5672 which are blocked by some networks.
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/README.md:222]: the proxy transport is available in the settings that can be specified in host.json.

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/README.md [hunk 3]
```diff
@@ -217,6 +217,39 @@ public static async Task Run(
...
+        // This method allows you to configure the options used to
+        // create the Service Bus client.
+        //
+        // In this example, we're setting the transport type to
+        // use web sockets so that traffic is directed over port 443
+        // instead of 5671 and 5672, which are blocked by some networks.
+
+        builder.AddServiceBus(options =>
+        {
+            options.TransportType = ServiceBusTransportType.AmqpWebSockets;
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/README.md:244]: It should be possible to configure this in host.json.

File: eng/Packages.Data.props [hunk 4]
```diff
@@ -197,6 +197,7 @@
     <PackageReference Update="Microsoft.Azure.Core.NewtonsoftJson" Version="1.0.0" />
     <PackageReference Update="Microsoft.Azure.Devices" Version="1.19.0" />
     <PackageReference Update="Microsoft.Azure.Devices.Client" Version="1.35.0" />
+    <PackageReference Update="Microsoft.Azure.Functions.Extensions" Version="1.1.0" />
```
Comment (JoshLove-msft) [inline eng/Packages.Data.props:200]: Why is this needed?

**Comment by JoshLove-msft** [comment]
It would be good to update this doc to show how to set the transport - https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-service-bus#additional-settings-for-version-5x

**Comment by jsquire** [comment]
Per conversations with Josh, this is actually available in the host settings which is a far more accessible way of setting it.   Closing this out, as it is not needed and will look to update the docs instead.
